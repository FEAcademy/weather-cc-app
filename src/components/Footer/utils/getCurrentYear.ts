const getCurrentYear = (): number => {
    const currentYear = new Date().getFullYear()
    return currentYear
}

export { getCurrentYear }
