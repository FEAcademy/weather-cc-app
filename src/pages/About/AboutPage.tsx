import mapImg from 'assets/images/map.png';
import parametersWidgetImg from 'assets/images/parametersWidget.png';
import temperatureWidgetImg from 'assets/images/temperatureWidget.png';
import { MainHeading, LogoContainer, SubHeading, RowContainer, Title, Description, Content } from './AboutPage.styled';
import { AboutPageTestIds } from './AboutPageTestIds';

const aboutPageData = [
  {
    title: 'WEATHER IN YOUR LOCATION',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus.',
    img: temperatureWidgetImg,
  },
  {
    title: 'AIR QUALITY',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus.',
    img: parametersWidgetImg,
  },
  {
    title: 'WEATHER ON MAP',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus.',
    img: mapImg,
  },
];

const AboutPage = () => (
  <div data-testid={AboutPageTestIds.Container}>
    <LogoContainer>
      <MainHeading>
        WEATHER<span>LY</span>
      </MainHeading>
      <SubHeading>weather app from CC frontend academy attendees</SubHeading>
    </LogoContainer>
    <Content>
      {aboutPageData.map(({ title, description, img }) => (
        <RowContainer key={title}>
          <img src={img} alt={title} />
          <div>
            <Title>{title}</Title>
            <Description>{description}</Description>
          </div>
        </RowContainer>
      ))}
    </Content>
  </div>
);

export { AboutPage };
