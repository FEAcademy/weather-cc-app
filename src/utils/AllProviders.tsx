import { PropsWithChildren } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { ThemeProvider } from '../theme/ThemeProvider';

const queryClient = new QueryClient();

const AllProviders = ({ children }: PropsWithChildren) => (
  <QueryClientProvider client={queryClient}>
    <ThemeProvider>{children}</ThemeProvider>
  </QueryClientProvider>
);

export { AllProviders };
