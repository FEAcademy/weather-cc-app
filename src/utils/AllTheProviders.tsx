import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import { ThemeProvider } from '../theme/ThemeProvider';

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  const history = createMemoryHistory();

  return (
    <ThemeProvider>
      <Router location={history.location} navigator={history}>
        {children}
      </Router>
    </ThemeProvider>
  );
};

export { AllTheProviders };
