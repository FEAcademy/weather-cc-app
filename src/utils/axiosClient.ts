import axios from 'axios';
import { IWeatherData } from './IWeatherData';

const apiClient = axios.create({
  baseURL: 'http://api.weatherapi.com/v1',
  params: {
    key: process.env.REACT_APP_WEATHER_API_KEY,
    aqi: 'yes',
  },
});

const getWeatherInCity = async (city: string): Promise<IWeatherData> => {
  const response = await apiClient.get<IWeatherData>('/current.json', { params: { q: city } });
  console.log({ response });
  return response.data;
};

export { getWeatherInCity };
