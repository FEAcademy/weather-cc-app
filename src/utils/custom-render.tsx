import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from '../theme/ThemeProvider';

const AllProvidersWrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <BrowserRouter>
      <ThemeProvider>{children}</ThemeProvider>
    </BrowserRouter>
  );
};

const renderWithRouter = (ui: ReactElement, { route = '/' } = {}, options?: Omit<RenderOptions, 'wrapper'>) => {
  window.history.pushState({}, 'Test page', route);

  return {
    ...render(ui, { wrapper: AllProvidersWrapper, ...options }),
  };
};

export * from '@testing-library/react';
export { renderWithRouter as render };
