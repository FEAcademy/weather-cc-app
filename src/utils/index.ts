export * from './custom-render';
export { Paths } from './Paths';
