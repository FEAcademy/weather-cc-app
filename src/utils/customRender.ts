import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
//import { BrowserRouter } from 'react-router-dom';

import { AllTheProviders } from './AllTheProviders';

const renderWithRouter = (ui: ReactElement, { route = '/' } = {}, options?: Omit<RenderOptions, 'wrapper'>) => {
  window.history.pushState({}, 'Test page', route);

  return {
    ...render(ui, { wrapper: AllTheProviders, ...options }),
  };
};

// const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
//   render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { renderWithRouter as render };
