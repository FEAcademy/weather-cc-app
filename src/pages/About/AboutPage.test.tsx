import { render, screen } from '../../utils/custom-render';
import { AboutPage } from './AboutPage';

describe('About page', () => {
  it('should render', () => {
    render(<AboutPage />);
    expect(screen.getByText(/Aboutpage/i)).toBeInTheDocument();
  });
});
