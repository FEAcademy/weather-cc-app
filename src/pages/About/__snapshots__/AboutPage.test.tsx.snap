// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="about-page-container"
  >
    <div
      class="sc-bczRLJ cLixUD"
    >
      <h1
        class="sc-gsnTZi ftbYjt"
      >
        WEATHER
        <span>
          LY
        </span>
      </h1>
      <h5
        class="sc-dkzDqf dfCfhD"
      >
        weather app from CC frontend academy attendees
      </h5>
    </div>
    <div
      class="sc-hKMtZM WXYlM"
    >
      <div
        class="sc-gKXOVf cmkqxb"
      >
        <img
          alt="WEATHER IN YOUR LOCATION"
          class="sc-iBkjds boxHpE"
          src="temperatureWidget.png"
        />
        <div>
          <h3
            class="sc-eCYdqJ flbLFE"
          >
            WEATHER IN YOUR LOCATION
          </h3>
          <p
            class="sc-jSMfEi coUyhO"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus.
          </p>
        </div>
      </div>
      <div
        class="sc-gKXOVf cmkqxb"
      >
        <img
          alt="AIR QUALITY"
          class="sc-iBkjds boxHpE"
          src="parametersWidget.png"
        />
        <div>
          <h3
            class="sc-eCYdqJ flbLFE"
          >
            AIR QUALITY
          </h3>
          <p
            class="sc-jSMfEi coUyhO"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus.
          </p>
        </div>
      </div>
      <div
        class="sc-gKXOVf cmkqxb"
      >
        <img
          alt="WEATHER ON MAP"
          class="sc-iBkjds boxHpE"
          src="map.png"
        />
        <div>
          <h3
            class="sc-eCYdqJ flbLFE"
          >
            WEATHER ON MAP
          </h3>
          <p
            class="sc-jSMfEi coUyhO"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
